payment/src/main/java/org/killbill/billing/payment/invoice/InvoicePaymentControlPluginApi.java

```diff
@@ -324,14 +324,15 @@ private UUID getInvoiceId(final Iterable<PluginProperty> pluginProperties) throw
     private PriorPaymentControlResult getPluginPurchaseResult(final PaymentControlContext paymentControlPluginContext, final Iterable<PluginProperty> pluginProperties, final InternalCallContext internalContext) throws PaymentControlApiException {
         try {
             final UUID invoiceId = getInvoiceId(pluginProperties);
-            final Invoice invoice = getAndSanitizeInvoice(invoiceId, paymentControlPluginContext.getAttemptPaymentId(), internalContext);
 
-            if (!InvoiceStatus.COMMITTED.equals(invoice.getStatus())) {
+            final InvoiceStatus status = invoiceApi.getInvoiceStatus(invoiceId, internalContext);
```

Comment:
Could you add a comment as to why we did this? I want to make sure future us don't _cleanup_ the optimization away 😆

Reply:
Amended the comment as requested.